include/pybind11/pybind11.h

```diff
@@ -2376,6 +2375,9 @@ inline function get_overload(const T *this_ptr, const char *name) {
 
 PYBIND11_NAMESPACE_END(PYBIND11_NAMESPACE)
 
+//
+// THE `pop` HERE NEED TO BE KEPT IN SYNC WITH THE CORRESPONDING `push` AT THE TOP OF THIS FILE.
```

Comment:
```suggestion
// THE `pop` HERE NEEDS TO BE KEPT IN SYNC WITH THE CORRESPONDING `push` AT THE TOP OF THIS FILE.
```

IMO "push" implies "pop", otherwise you wouldn't bother to "push" in the first place, but not horribly against this note if you want it.

Reply:
I removed the notes completely. Hopefully a few PRs down the road this block will be gone entirely!

Note that I started out wanting to fix the missing pop for Intel, but in the meantime the direction of this PR changed a lot and the notes don't make that much sense anymore.